feat: implement collections array with names for archived item details

Chickensoupwithrice opened the pull request:
Resolves the backend for https://github.com/webrecorder/browsertrix-cloud/issues/1070 by "annotating" the `collections` field with addition `name`. 

This changes the return value from an array of strings (`collections: [string, ...]`) to an array of objects (`collections: [{"name": string, "id": string}, ...]`)

It only addresses the GET request portion of that issue. PATCH / update to follow in a different PR

Deploying to dev now to see the results, tested in `mongodb-compass` appears to work! 

Shout outs Tessa for helping me ♥️

Chickensoupwithrice committed:
feat: implement collectionNames on base_crawls API

Chickensoupwithrice committed:
fix: format

Chickensoupwithrice committed:
fix: disable pylint too-many-statements for list_all_base_crawls

Chickensoupwithrice committed:
fix: annotate `collections` itself

This is instead of using a new key (`collectionNames`)

ikreymer commented:
It turns out that we actually want this implemented in the single `get_crawl()` path rather than list_* paths.
We *don't* want to change this for list, as that incurs an extra lookup (every crawl will look up every collection its in).
The list of collection ids we get for free since its part of the existing query.

A question is what should the output be, and we also want to be consistent.
There's a couple of options:
1) Change `collections` to be `(collections: [{"name": string, "id": string}, ...])` for single item/details page. For list pages, remove the `collections` key altogether since we don't want use same key as different formats. But then we lose data that we otherwise have 'for free', eg. 

2) Keep `collections` key as `collections: [string, ...]` on all responses. On single item/details add a `collectionNames: [string, ...]` which provides the names for the ids, but in a separate array. This adds the names, but uses two objects.

3) Keep `collections` key as `collections: [string, ...]` on all responses. On single item/details, add a `collectionsAndNames: [{"name": string, "id": string}, ...])` which includes both name and id. This has the benefit of 1), but has downside of including duplicate data.

@SuaYoo are there particular considerations for frontend that would lean us into one or the other here?

ikreymer committed:
move collection name resolution to single crawl from multiple crawl resolution
rename 'collections' -> 'collectionIds', adding migration
only populate 'collections' field with {name, id} pair for get_crawl() / single archived item
path, not for aggregate/list methods
remove Crawl.get_crawl(), redundant with BaseCrawl.get_crawl() version
ensure _files_to_resources returns an empty [] instead of none if empty (matching BaseCrawl.get_crawl() behavior to Crawl.get_crawl())
tests: update tests to use collectionIds for id list, add 'collections' for {name, id} test
frontend: change Crawl object to have collectionIds instead of collections

ikreymer commented:
Per discussion on discord, going with 3) but using `collectionIds` for id list and `collections` for collection detail, which may be expanded in the future.

ikreymer committed:
add missing migration file

ikreymer commented:
Looks like this should be ready for review, revamped original implementation with get_crawl() and collections -> collectionsId rename.

tw4l commented on frontend/src/types/crawler.ts:
```diff
@@ -126,7 +126,7 @@ export type Crawl = CrawlConfig & {
   firstSeed: string;
   seedCount: number;
   stopping: boolean;
-  collections: string[];
```
If we're going to be fetching names in `collections`, should that still be part of the type here, just with a different type signature, similar to resources?

tw4l replied:
Or maybe Sua will add that when we make use of the new collection names?

ikreymer replied:
yea, was thinking that'd be added as part of the frontend PR that is using them

tw4l reviewed (approved):
Looks good! Tested locally, including the migration, and all seems to be working as expected.
